sphinx_gherkindoc/glossary.py

```diff
@@ -33,12 +33,14 @@ def tuple_len(self) -> Tuple[int, int]:
         """Get the length for each location and the number of associated steps."""
         return (len(self.locations), sum(map(len, self.locations.values())))
 
-    def __gt__(self, other) -> bool:
+    def __gt__(self, other: "GlossaryEntry") -> bool:
         """Compare the location and step lenghts for the larger one."""
         return self.tuple_len() > other.tuple_len()
 
-    def __eq__(self, other) -> bool:
+    def __eq__(self, other: object) -> bool:
         """Compare the location and step lengths for equality."""
+        if not isinstance(other, GlossaryEntry):
+            return NotImplemented
```

Comment:
Curious why you needed to add this? Can't you just say `other: GlossaryEntry` and be done with it? Do we have places where we call this when the `other` isn't of that type?

Reply:
I started that way, but `mypy` barked at me for redefining `__eq__` args, which apparently must accept `other: Any`  then raise an exception for any non-supported types.

Reply:
Yup, all the dunder relational ops can take anything as the other value, and there are rules for subclass type checking as to which operand of `==` will be the to get called.
https://docs.python.org/3/reference/datamodel.html#object.__eq__